trin-core/src/portalnet/types/messages.rs

```diff
@@ -619,272 +570,205 @@ impl FromStr for HexData {
-        let message = Message::Response(Response::Nodes(Nodes { total, enrs }));
-        assert_eq!(hex::encode(message.to_bytes()), expected);
-        let decoded = Message::from_bytes(message.to_bytes().as_slice()).unwrap();
-        assert_eq!(decoded, message);
-    }
+        let encoded: Vec<u8> = content.clone().into();
+        let encoded = hex::encode(encoded);
+        let expected_encoded = "05000102";
+        assert_eq!(encoded, expected_encoded);
 
-    #[test]
-    fn test_vector_find_content() {
-        let content_key = hex::decode("706f7274616c").unwrap();
-        let expected = "0404000000706f7274616c";
-        let message = Message::Request(Request::FindContent(FindContent { content_key }));
-        assert_eq!(hex::encode(message.to_bytes()), expected);
-        let decoded = Message::from_bytes(message.to_bytes().as_slice()).unwrap();
-        assert_eq!(decoded, message);
+        let decoded = Message::try_from(hex::decode(expected_encoded).unwrap()).unwrap();
+        assert_eq!(decoded, content);
     }
 
     #[test]
-    fn test_vector_content_with_connection_id() {
-        let raw = [01u8, 02u8];
-        let connection_id = u16::from_le_bytes(raw);
-        let expected = "05000102";
-        let message = Message::Response(Response::Content(Content::ConnectionId(connection_id)));
-        assert_eq!(hex::encode(message.to_bytes()), expected);
-        let decoded = Message::from_bytes(message.to_bytes().as_slice()).unwrap();
-        assert_eq!(decoded, message);
-    }
+    fn message_encoding_content_content() {
+        let content_val = hex::decode("7468652063616b652069732061206c6965").unwrap();
+        let content_val = ByteList::from(VariableList::from(content_val));
+        let content = Content::Content(content_val);
+        let content = Message::Content(content);
 
-    #[test]
-    fn test_vector_content_with_content() {
-        let content = ByteList::from(VariableList::from(
-            hex::decode("7468652063616b652069732061206c6965").unwrap(),
-        ));
-        let expected = "05017468652063616b652069732061206c6965";
-        let message = Message::Response(Response::Content(Content::Content(content)));
-        assert_eq!(hex::encode(message.to_bytes()), expected);
-        let decoded = Message::from_bytes(message.to_bytes().as_slice()).unwrap();
-        assert_eq!(decoded, message);
+        let encoded: Vec<u8> = content.clone().into();
+        let encoded = hex::encode(encoded);
+        let expected_encoded = "05017468652063616b652069732061206c6965";
+        assert_eq!(encoded, expected_encoded);
+
+        let decoded = Message::try_from(hex::decode(expected_encoded).unwrap()).unwrap();
+        assert_eq!(decoded, content);
     }
 
     #[test]
-    fn test_vector_content_with_enrs() {
+    fn message_encoding_content_enrs() {
```

Comment:
It might be a hangover from python - but I prefix every test with `test_*` - seems like it's not really a thing in Rust. I guess given the `#[test]` decorator, it's obviously a test. But on the other hand I do find it useful to use the `test_` prefix when grepping through a file. Whatever we decide on should probably be added to the `CONTRIBUTING.md` doc so the codebase stays consistent.

Reply:
Yeah, it's just my personal preference to omit the `test` prefix due to redundancy. It's not a strong preference, but in the interest of consistency, I can add a note to `CONTRIBUTING.md` and perhaps a follow-up PR to refactor any existing tests. If you have further thoughts, please let me know. If not, I'll assume that we will go with this convention.